packages/block-library/src/query/edit/inspector-controls/parent-control.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { FormTokenField } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { useState, useEffect, useMemo } from '@wordpress/element';
+import { useDebounce } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { getEntitiesInfo, mapToIHasNameAndId } from '../../utils';
+
+const EMPTY_ARRAY = [];
+const SUGGESTIONS_QUERY = {
+	per_page: 20,
+	order: 'asc',
+	orderby: 'title',
+	_fields: 'id,title',
+	context: 'view',
+};
+
+function ParentControl( { parents, postType, onChange } ) {
+	const [ search, setSearch ] = useState( '' );
+	const [ value, setValue ] = useState( EMPTY_ARRAY );
+	const [ suggestions, setSuggestions ] = useState( EMPTY_ARRAY );
+	const debouncedSearch = useDebounce( setSearch, 250 );
+	const { searchResults, searchHasResolved } = useSelect(
+		( select ) => {
+			if ( ! search ) {
+				return { searchResults: EMPTY_ARRAY, searchHasResolved: true };
+			}
+			const { getEntityRecords, hasFinishedResolution } = select(
+				coreStore
+			);
+			const selectorArgs = [
+				'postType',
+				postType,
+				{
+					...SUGGESTIONS_QUERY,
+					search,
+					exclude: parents,
+				},
+			];
+			return {
+				searchResults: getEntityRecords( ...selectorArgs ),
+				searchHasResolved: hasFinishedResolution(
+					'getEntityRecords',
+					selectorArgs
+				),
+			};
+		},
+		[ search, parents ]
+	);
+	const currentParents = useSelect(
+		( select ) => {
+			if ( ! parents?.length ) return EMPTY_ARRAY;
+			const { getEntityRecords } = select( coreStore );
+			return getEntityRecords( 'postType', postType, {
+				...SUGGESTIONS_QUERY,
+				include: parents,
```

Comment:
I think we still want to do an unbound or `per_page: parents.length` query here.

Reply:
What do you mean with `unbound`? 

`include` does limit the results to the specified ids (if exist).
>Limit result set to specific IDs.

Reply:
> What do you mean with unbound?

Sorry, I mean `per_page: -1`.

> include does limit the results to the specified ids (if exist).

If `per_page` is lower than items to include, it will limit the number of results. I hope people won't filter to include 20 parents 😄 

The following only returns a single item.

```
wp.data.select('core').getEntityRecords( 'postType', 'post', { include: [ 1, 17, 37 ], per_page: 1 } );
```

Reply:
Oh, great catch George! I updated in my last [commit](https://github.com/WordPress/gutenberg/pull/40933/commits/b36f63989b78df789bf234b70431fa91d5fb2369).